[Python 3.12] `ModuleNotFoundError: No module named 'setuptools'` as `setuptools` not in the `venv`, only installed by `apt`!

Failing CI job: https://github.com/darktable-org/rawspeed/actions/runs/9698273466/job/26764443826
`python3-setuptools` package is installed: https://github.com/darktable-org/rawspeed/actions/runs/9698273466/job/26764443826#step:5:765

`make venv` does not seem to remove it:
https://github.com/darktable-org/rawspeed/actions/runs/9698273466/job/26764443826#step:10:17

Yet `make package` ultimately fails:
https://github.com/darktable-org/rawspeed/actions/runs/9698273466/job/26764443826#step:10:250
```
+ BUILD_LOGGER_64_BIT_ONLY=YES BUILD_UI_DIST=NO make package
mkdir -p /__w/rawspeed/rawspeed/codechecker/build && \
mkdir -p /__w/rawspeed/rawspeed/codechecker/build/CodeChecker/bin && \
mkdir -p /__w/rawspeed/rawspeed/codechecker/build/CodeChecker/lib/python3
if [ -d "/__w/rawspeed/rawspeed/codechecker/.git" ]; then git config --local commit.template .gitmessage; fi
cp -p scripts/gerrit_changed_files_to_skipfile.py /__w/rawspeed/rawspeed/codechecker/build/CodeChecker/bin
BUILD_DIR=/__w/rawspeed/rawspeed/codechecker/build BUILD_LOGGER_64_BIT_ONLY=YES make -C /__w/rawspeed/rawspeed/codechecker/analyzer package_analyzer
make[1]: Entering directory '/__w/rawspeed/rawspeed/codechecker/analyzer'
mkdir -p /__w/rawspeed/rawspeed/codechecker/build && \
mkdir -p /__w/rawspeed/rawspeed/codechecker/build/CodeChecker/bin && \
mkdir -p /__w/rawspeed/rawspeed/codechecker/build/CodeChecker/lib/python3
make -C /__w/rawspeed/rawspeed/codechecker/analyzer/../tools/tu_collector build
make[2]: Entering directory '/__w/rawspeed/rawspeed/codechecker/tools/tu_collector'
python3 setup.py build --build-purelib build/tu_collector
Traceback (most recent call last):
  File "/__w/rawspeed/rawspeed/codechecker/tools/tu_collector/setup.py", line 4, in <module>
    import setuptools
ModuleNotFoundError: No module named 'setuptools'
make[2]: *** [Makefile:44: build] Error 1
make[2]: Leaving directory '/__w/rawspeed/rawspeed/codechecker/tools/tu_collector'
make[1]: *** [Makefile:54: build_tu_collector] Error 2
make[1]: Leaving directory '/__w/rawspeed/rawspeed/codechecker/analyzer'
make: *** [Makefile:45: package] Error 2
Error: Process completed with exit code 2.
```

Google suggests that `setuptools` should not be in `requirements.txt` and certainly should not be in a lock file.
I'm a little bit lost as to what is going on.

X-Ref: https://github.com/darktable-org/rawspeed/pull/738#issuecomment-2194523426

--- a/analyzer/tools/merge_clang_extdef_mappings/requirements_py/dev/requirements.txt
+++ b/analyzer/tools/merge_clang_extdef_mappings/requirements_py/dev/requirements.txt
@@ -1,3 +1,4 @@
 pytest==7.3.1
 pycodestyle==2.12.0
 pylint==3.2.4
+setuptools==70.2.0

--- a/analyzer/tools/statistics_collector/requirements_py/dev/requirements.txt
+++ b/analyzer/tools/statistics_collector/requirements_py/dev/requirements.txt
@@ -1,3 +1,4 @@
 pytest==7.3.1
 pycodestyle==2.12.0
 pylint==3.2.4
+setuptools==70.2.0

--- a/codechecker_common/requirements_py/dev/requirements.txt
+++ b/codechecker_common/requirements_py/dev/requirements.txt
@@ -3,3 +3,4 @@ coverage==5.5.0
 mypy==1.7.1
 PyYAML==6.0.1
 types-PyYAML==6.0.12.12
+setuptools==70.2.0

--- a/tools/bazel/requirements_py/dev/requirements.txt
+++ b/tools/bazel/requirements_py/dev/requirements.txt
@@ -2,3 +2,4 @@ pytest==7.3.1
 pycodestyle==2.12.0
 pylint==3.2.4
 mypy==1.7.1
+setuptools==70.2.0

--- a/tools/report-converter/requirements_py/dev/requirements.txt
+++ b/tools/report-converter/requirements_py/dev/requirements.txt
@@ -4,3 +4,4 @@ pycodestyle==2.12.0
 pylint==3.2.4
 portalocker==2.2.1
 mypy==1.7.1
+setuptools==70.2.0

--- a/tools/tu_collector/requirements_py/dev/requirements.txt
+++ b/tools/tu_collector/requirements_py/dev/requirements.txt
@@ -2,3 +2,4 @@ pytest==7.3.1
 pycodestyle==2.12.0
 pylint==3.2.4
 mypy==1.7.1
+setuptools==70.2.0

